src/mono/mono/mini/monovm.c

```diff
@@ -118,7 +118,7 @@ mono_core_preload_hook (MonoAssemblyLoadContext *alc, MonoAssemblyName *aname, c
 	for (int i = 0; i < a->assembly_count; ++i) {
 		if (!strcmp (basename, a->basenames[i])) {
 			MonoAssemblyOpenRequest req;
-			mono_assembly_request_prepare_open (&req, refonly ? MONO_ASMCTX_REFONLY : MONO_ASMCTX_DEFAULT, default_alc);
+			mono_assembly_request_prepare_open (&req, MONO_ASMCTX_DEFAULT, default_alc);
```

Comment:
should we have

```c
if (refonly)
  return NULL;
```

somewhere near the beginning of this hook?

Reply:
I'd rather assert if we're going to go that route. If this is passed refonly or installed on legacy Mono, something is very wrong.

Reply:
I'll add that, given I'm going to have to push again anyway to fix Windows builds...